### DISCUSSION TITLE
Online Access

### DISCUSSION THREAD

**Discussion by hdg630** [zauberzeug/nicegui#1407] [Q&A]
Is there a way to give the created gui a url that can be accessed online instead of only locally? 

Right now my gui is only accessible on http://localhost:8080/

i.e. I made a gui and I want friends to be able to access it as well.

**Comment by rodja** [comment]
Thanks for asking. We have been working on this for quite a while. You can use `ui.run(on_air=True)` to make your local app remotely accessible. See [NiceGUI On Air](https://nicegui.io/documentation#nicegui_on_air) for more informations.

**Reply by hdg630** [reply]
Thank you for the reply. Is there a way to do this without having the python file running? Or is there a tutorial out there to be able to turn this into a website?

**Reply by rodja** [reply]
> Is there a way to do this without having the python file running?

If you want to see the GUI you need to have it running somewhere. That can be local or on a server. But the Python code needs to be executed somewhere.

>  Or is there a tutorial out there to be able to turn this into a website?

It is a website. When starting a NiceGUI app you start a webserver which "serves your dynamic website". Export to plain html/css/js is not possible. See https://github.com/zauberzeug/nicegui/discussions/313.
